examples/tests/lr1110/lorawan/main_lorawan.c

```diff
@@ -0,0 +1,460 @@
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Semtech corporation nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE GRANTED BY
+ * THIS LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SEMTECH CORPORATION BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- DEPENDENCIES ------------------------------------------------------------
+ */
+
+// #include "main_lorawan.h"
+// #include "lorawan_key_config.h"
+// #include "smtc_board.h"
+// #include "smtc_hal.h"
+// #include "apps_modem_common.h"
+// #include "apps_modem_event.h"
+// #include "smtc_modem_api.h"
+// #include "device_management_defs.h"
+// #include "smtc_board_ralf.h"
+// #include "apps_utilities.h"
+// #include "smtc_modem_utilities.h"
+
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include <libtock-sync/sensors/humidity.h>
+#include <libtock-sync/sensors/temperature.h>
+#include <libtock-sync/services/alarm.h>
+
+#include <lr1110/lr1110.h>
+#include <lr1110/us_915_ttn.h>
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PRIVATE MACROS-----------------------------------------------------------
+ */
+
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PRIVATE CONSTANTS -------------------------------------------------------
+ */
+
+#define LORAWAN_REGION      SMTC_MODEM_REGION_US_915
+#define LORAWAN_CLASS       SMTC_MODEM_CLASS_A
+
+// wm1110dev parameters
+#define LORAWAN_DEVICE_EUI "70B3D57ED00650D9"
+#define LORAWAN_JOIN_EUI   "901AB1F40E1BCC81"
+#define LORAWAN_APP_KEY    "3356A7047ECF1F2F78C72AE9B1635BC1"
+
+/*!
+ * @brief Defines the application data transmission duty cycle. 60s (changed to 3s), value in [s].
+ */
+#define APP_TX_DUTYCYCLE 3
```

Comment:
I remembered it is how often the app makes an attempt to transmit data to TTN, and if the transmission succeeds, the duty cycle will stop. I agree that we should make it a larger number, but if it's too large, like 60, it might fail the app.

Reply:
It's way too low then, this needs to be at least 60 minutes

Reply:
Since this is only for a test app, the impact should be minimal here.

Reply:
It's not minimal impact, people will be banned from TTN if they run the test app and get caught.

Not to mention it's possible a crime in some countries